test/check-machines-create

```diff
@@ -313,7 +313,8 @@ class TestMachinesCreate(VirtualMachinesCase):
         # Redefine the domain with the new XML
         xmlstr = ET.tostring(root, encoding='unicode', method='xml')
 
-        self.machine.execute(f"echo '{xmlstr}' > /tmp/domain.xml && virsh define --file /tmp/domain.xml")
+        self.write_file("/tmp/domain.xml", xmlstr)
+        self.machine.execute("virsh define --file /tmp/domain.xml")
```

Comment:
So the problem here, if I understand it correctly was that virsh define was reading the old file domain.xml?

Reply:
No, /tmp/domain.xml has always been okay. But the running virt-install in the background and this `virsh define` collided, and sometimes virt-install won -- hence this `virsh define` was a no-op.